Client/Frontend/Home/HomepageViewController.swift

```diff
@@ -755,8 +755,8 @@ extension HomepageViewController: HomepageViewModelDelegate {
             guard let self = self else { return }
 
             self.viewModel.refreshData(for: self.traitCollection, size: self.view.frame.size)
-            self.collectionView.reloadData()
             self.collectionView.collectionViewLayout.invalidateLayout()
+            self.collectionView.reloadData()
```

Comment:
reloadData should come first as invalidateLayout causes the compositional layout block to be invoked, and if it is run before reloadData then it can result in a mismatch between the actual data and the expected data for the layout.

Reply:
I've made this change first but it was causing the crash on iOS 16. I can revert because I think the main cause of the crash is the recent tabs not being in sync with the adaptor data source